shell/browser/browser_mac.mm

```diff
@@ -456,12 +456,17 @@ void RemoveFromLoginItems() {
 }
 
 void Browser::DockSetIcon(v8::Isolate* isolate, v8::Local<v8::Value> icon) {
-  api::NativeImage* native_image = nullptr;
-  if (!api::NativeImage::TryConvertNativeImage(isolate, icon, &native_image))
-    return;
+  gfx::Image image;
+
+  if (!icon->IsNull()) {
+    api::NativeImage* native_image = nullptr;
+    if (api::NativeImage::TryConvertNativeImage(isolate, icon, &native_image)) {
+      image = native_image->image();
+    }
```

Comment:
Best to not do anything if `TryConvertNativeImage` returns false, as it will have thrown an exception.

```suggestion
    if (!api::NativeImage::TryConvertNativeImage(isolate, icon, &native_image))
      return;
    image = native_image->image();
```